lib/PreviewProvider.js

```diff
@@ -277,15 +219,26 @@ PreviewProvider.prototype = {
   }),
 
   /**
+   * Find the metadata for each link in the database
+   */
+  _asyncFindItemsInDB: Task.async(function*(links) {
+    const cacheKeyArray = links.map(link => link.cache_key);
+    let items = yield this._metadataStore.asyncGetMetadataByCacheKey(cacheKeyArray);
```

Comment:
Try to avoid using variable names like 'items' or 'things' or 'puppies'.  Describe what the thing actually is, in this case I think it's a bunch of links with their metadata?  linksMetadata?  Also I think this whole function is:

return yield this._metadataStore.asyncGetMetadataByCacheKey(links.map(link => link.cache_key));